pkg/reporting/cookbooks.go

```diff
@@ -196,58 +182,60 @@ func (cbr *CookbooksReport) addRecord(r *CookbookRecord) {
 	cbr.Records = append(cbr.Records, r)
 }
 
-func (cbr *CookbooksReport) triggerJobs(cookbooks chef.CookbookListResult, inCh chan<- cookbookItem) {
-	for cookbookName, cookbookVersions := range cookbooks {
+func (cbr *CookbooksReport) triggerJobs(inCh chan<- cookbookItem) {
+	for cookbookName, cookbookVersions := range cbr.cookbookSearchResults {
 		for _, ver := range cookbookVersions.Versions {
 			inCh <- cookbookItem{cookbookName, ver.Version}
 		}
 	}
-	//debug("%s: finished sending jobs", time.Now())
 	close(inCh)
 }
 
 func (cbr *CookbooksReport) createDownloadWorkerPool(nWorkers int, downloadCh <-chan cookbookItem, analyzeCh chan<- *CookbookRecord) {
 	var wg sync.WaitGroup
-
 	for i := 0; i < nWorkers; i++ {
 		wg.Add(1)
 		go func(inCh <-chan cookbookItem, outCh chan<- *CookbookRecord, wg *sync.WaitGroup) {
 			for item := range inCh {
-				cbr.downloadCookbook(item.Name, item.Version, outCh)
+				cbState := cbr.downloadCookbook(item.Name, item.Version)
+				analyzeCh <- cbState
 			}
 			wg.Done()
+
 		}(downloadCh, analyzeCh, &wg)
 	}
 
 	wg.Wait()
-	//debug("%s: finished downloading", time.Now())
```

Comment:
I still feel we should create a debugging function to print useful debug information. 😅 But if you think we will remember where to add debug messages, then Im fine removing this since It is more like a reminder for me ha.